includes/Admin/Editor.php

```diff
@@ -358,6 +359,44 @@ public function get_editor_settings(): array {
 		return apply_filters( 'web_stories_editor_settings', $settings );
 	}
 
+	/**
+	 * Get taxonomy endpoints.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @return array
+	 */
+	protected function get_taxonomy_endpoints() : array {
```

Comment:
FYI, I don't think we need this.

The editor fetches all taxonomies from `web-stories/v1/taxonomies` and then gets the endpoints for each taxonomy from the `_links` in the REST API response, as intended by HAL.

Reply:
For context, I have added this here, as the endpoints are different from the core ones. So /web-stories/v1 and not wp/v2. Just ensure we are connecting to the correct urls. I will removed if not needed.

CC @littlemilkstudio @BrittanyIRL

Reply:
I know. But the editor doesn't use this config.

What's needed is that the `web-stories/v1/taxonomies` endpoint, which lists all taxonomies, points to all the `web-stories/v1` endpoints for each individual taxonomy in the `_links` section. In the editor, all taxonomy endpoints are derived from `_links`.

To achieve this:

1. `git merge main` so you get `Stories_Taxonomies_Controller` in your branch
2. Add the following method to `Stories_Taxonomies_Controller` (might need tweaking):

```php
/**
 * Prepares a taxonomy object for serialization.
 *
 * @since 1.12.0
 *
 * @param WP_Taxonomy     $taxonomy Taxonomy data.
 * @param WP_REST_Request $request  Full details about the request.
 * @return WP_REST_Response Response object.
 */
public function prepare_item_for_response( $taxonomy, $request ) {
	$response = parent::prepare_item_for_response( $taxonomy, $request );

	$base = ! empty( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxonomy->name;

	$response->remove_link( 'https://api.w.org/items' );
	$response->add_links(
		[
			'https://api.w.org/items' => [
				'href' => rest_url( sprintf( 'web-stories/v1/%s', $base ) ),
			],
		]
	);

	/** This filter is documented in wp-includes/rest-api/endpoints/class-wp-rest-taxonomies-controller.php */
	return apply_filters( 'rest_prepare_taxonomy', $response, $taxonomy, $request );
}
```

Then, visit `wp-json/web-stories/v1/taxonomies?type=web-story` and you'll see the correct links as expected:

![Screenshot 2021-09-17 at 11 08 53](https://user-images.githubusercontent.com/841956/133757103-7c1b4e72-e2c5-4e3c-b398-6af73770057f.png)

That's all that's needed in the editor and you can remove this `get_taxonomy_endpoints()` code completely.